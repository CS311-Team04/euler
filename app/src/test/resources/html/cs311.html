<!DOCTYPE html>
<html lang="en" class="tablesaw-enhanced">

<head>
<meta name="isa_raw_url" content="http://isa.epfl.ch/imoniteur_ISAP/!itffichecours.htm?ww_i_matiere=3517600663&ww_x_anneeAcad=2025-2026&ww_i_section=249847&ww_i_niveau=6683117&ww_c_langue=en">        <meta name="description" content="This course teaches the journey taken by software engineering teams from incipient ideas to software products that solve real problems for real people.">
        <link rel="canonical" href="https://edu.epfl.ch/coursebook/en/the-software-enterprise-from-ideas-to-products-CS-311">
<title>The software enterprise - from ideas to products - CS-311 - EPFL</title>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/x-icon" href="https://web2018.epfl.ch/8.1.0/icons/favicon.ico">
<link rel="icon" type="image/png" sizes="16x16" href="https://web2018.epfl.ch/8.1.0/icons/favicon-16.png">
<link rel="icon" type="image/png" sizes="32x32" href="https://web2018.epfl.ch/8.1.0/icons/favicon-32.png">
<link rel="apple-touch-icon" sizes="180x180" href="https://web2018.epfl.ch/8.1.0/icons/apple-touch-icon.png">

<link rel="stylesheet" href="https://web2018.epfl.ch/8.1.0/css/elements.min.css">

<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-PJGBG5R');</script>
    <link rel="stylesheet" href="/assets/css/graph-search-widget.css">
</head>


<body>

<div id="app" class="site">
<header role="banner" class="header header-light">
    <div class="header-light-content">
        <a class="logo" href="//www.epfl.ch">
            <img src="https://web2018.epfl.ch/8.1.0/icons/epfl-logo.svg"
                 alt="Logo EPFL, École polytechnique fédérale de Lausanne"
                 class="img-fluid">
        </a>

        <p class="site-title">
            <a href="/studyplan/en">Coursebooks</a>
        </p>

        <!-- search -->
        <form action="//search.epfl.ch" class="d-xl-none">
            <a id="search-mobile-toggle" class="search-mobile-toggle searchform-controller" href="#">
                <svg class="icon" aria-hidden="true">
                    <use xlink:href="#icon-search"></use>
                </svg>
                <span class="toggle-label sr-only">Show / hide the search form</span>
            </a>
            <div class="input-group search-mobile" role="search">

                <div class="input-group-prepend">
                        <span class="input-group-text">
                            <svg class="icon" aria-hidden="true">
                                <use xlink:href="#icon-search"></use>
                            </svg>
                        </span>
                </div>
                <label for="q-mobile" class="sr-only">Search</label>
                <input type="text" class="form-control" id="q-mobile" name="q" placeholder="Search">
                <div class="input-group-append">
                    <a id="search-mobile-close" class="search-mobile-close searchform-controller" href="#">
                        <svg class="icon" aria-hidden="true">
                            <use xlink:href="#icon-close"></use>
                        </svg>
                        <span class="toggle-label sr-only">Hide the search form</span>
                    </a>
                </div>
            </div>
        </form>
        <div class="dropdown dropright search d-none d-xl-block">
            <a class="dropdown-toggle" href="#" data-toggle="dropdown" title="Search">
                <svg class="icon" aria-hidden="true">
                    <use xlink:href="#icon-search"></use>
                </svg>
            </a>
            <form action="//search.epfl.ch" class="dropdown-menu border-0 p-0">
                <div class="search-form mt-1 input-group">
                    <label for="q" class="sr-only">Search</label>
                    <input type="text" class="form-control" id="q" name="q" placeholder="Search...">
                    <button type="submit" class="d-none d-xl-block btn btn-primary input-group-append">Search
                    </button>
                </div>
            </form>
        </div>

        <nav class="nav-lang nav-lang-short ml-auto">
            <ul><li><a aria-label="Français" href="/coursebook/fr/the-software-enterprise-from-ideas-to-products-CS-311">FR</a></li><li><span aria-label="English" class="active">EN</span></li></ul>
        </nav>
    </div>
</header>
    <!-- body -->
    <div class="main-container">
        <!-- breadcrumb -->
<div class="breadcrumb-container">
    <nav aria-label="breadcrumb" class="breadcrumb-wrapper">
        <ol class="breadcrumb pb-0">
            <li class="breadcrumb-item">
                <a href="//www.epfl.ch" title="Go to main site" aria-label="Go to main site" class="bread-link bread-home">
                    <svg class="icon" aria-hidden="true">
                        <use xlink:href="#icon-home"></use>
                    </svg>
                </a>
            </li>
            <li class="breadcrumb-item expand-links">
                <button class="btn btn-expand-links" aria-expanded="false" title="Display the complete path">
                    <span class="dots" aria-hidden="true">…</span>
                    <span class="sr-only">Display the complete path</span>
                </button>
            </li>
                    <li class="breadcrumb-item">
                        <a href="/index.en.html">
                            Study plans
                        </a>
                    </li>
                    <li class="breadcrumb-item">
                        <a href="">
                            Coursebooks
                        </a>
                    </li>
                    <li class="breadcrumb-item active">
                        The software enterprise - from ideas to products
                    </li>
        </ol>
    </nav>
</div>
        <!-- main container -->
        <div class="nav-toggle-layout nav-aside-layout">
            <div class="w-100 pb-3">
                <main id="main" role="main" class="content container-grid">
                    <div class="container mt-5">
                        <header class="page-header d-xl-flex justify-content-xl-between align-items-xl-center mb-5">
                            <h1 class="mb-4 mb-xl-0">The software enterprise - from ideas to products</h1>

                            <div class="mr-0">
                            <a href="https://isa.epfl.ch/imoniteur_ISAP/!GEDPUBLICREPORTS.pdf?ww_i_reportModel=1696552884&ww_i_reportModelXsl=1696552963&ww_i_itemplan=4404062591&ww_c_langue=en">
                                <button class="btn btn-sm btn-secondary">
                                    <svg class="icon feather mb-1" aria-hidden="true">
                                        <use xlink:href="#download"></use>
                                    </svg>
                                    Download the coursebook (PDF)
                                </button>
                            </a>
                            </div>
                        </header>

                        <div class="row">
                            <div class="course-details col-xl-7">
                                <div class="course-summary mb-5">
                                    <p>CS-311 / <strong>8 credits</strong></p>
                                    <p><strong>Teacher: </strong> <a href='https://people.epfl.ch/172241?lang=en'>Candea George</a></p>
                                    <p><strong>Language:</strong> English</p>
                                        <p><strong>Withdrawal:</strong> It is not allowed to withdraw from this subject after the registration deadline.</p>
                                        <p><strong>Remark:</strong> Special schedule</p>
                                </div>
                                <hr>
                                    <div class="mt-5">
                                                <h2 class="h5">Summary</h2>
                                                <p>This course teaches the journey taken by software engineering teams from incipient ideas to software products that solve real problems for real people.</p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Content</h2>
                                            <div class="collapse collapse-partial" id="course-content">
                                                <p><p>The skills acquired in this course will enable students to build effective software products in teams, either within an existing organization or as founders of their own startups.</p> <ul>
<li>Requirements and specifications</li>
<li>Validation, testing, and debugging</li>
<li>DevOps (version control, project management, issue tracking, continuous integration)</li>
<li>Behavior-driven and test-driven development</li>
<li>Development processes</li>
<li>Cloud-platform and mobile-platform architectures</li>
<li>Product architecture</li>
<li>Security, privacy, and data protection</li>
</ul></p>
                                            </div>
                                            <button class="collapsed btn btn-light btn-block btn-sm mt-3" aria-expanded="false" data-toggle="collapse" data-target="#course-content" aria-controls="course-content">
                                                Show more
                                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-triangle"></use></svg>
                                            </button>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Learning Prerequisites</h2>
                                                <h2 class="h6">Required courses</h2>
                                                <p><ul>
<li>CS-214 Software construction (BA3)</li>
<li>CS-202 Computer systems (BA4)</li>
</ul></p>
                                                <h2 class="h6">Important concepts to start the course</h2>
                                                <p><p>Must be proficient in programming in C and Python and object-oriented Java/Kotlin/Scala</p></p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Learning Outcomes</h2>
                                                <p>By the end of the course, the student must be able to:<br></p>
                                                <p><ul><li>Design and implement mobile and/or cloud apps</li><li>Master a variety of system design patterns</li><li>Work in (and manage) a team of developers</li><li>Identify opportunities for using software to solve real-world problems</li><li>Plan a software product from A-to-Z</li><li>Assess / Evaluate progress against the plan, and adapt the plan as appropriate</li><li>Manage priorities + master the basics of product management</li><li>Optimize the use of time and resources to achieve a given goal</li><li>Take feedback (critique) and respond in an appropriate manner</li><li>Develop  auto-didact skills</li></ul></p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Teaching methods</h2>
                                                <p><ul>
<li>Ex cathedra</li>
<li>Extensive team-based project</li>
</ul></p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Expected student activities</h2>
                                                <p><ul>
<li>Work with team members to complete a substantial project</li>
<li>Independently research solutions, study documentation, etc. (auto-didact)</li>
</ul></p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Assessment methods</h2>
                                                <p><ul>
<li>Throughout the semester (continuous control)</li>
<li>Grade determined based on both team and individual performance in the project</li>
<li>Deliverables include an implemented software product</li>
</ul></p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Supervision</h2>
                                                <p><table><tbody><tr><td>
Office hours
      </td><td>
Yes
      </td></tr><tr><td>
Assistants
      </td><td>
Yes
      </td></tr></tbody></table></p>
                                    </div>
                                    <div class="mt-5">
                                                <h2 class="h5">Resources</h2>
                                                <h2 class="h6">Websites</h2>
                                                <p><ul><li><link><a href="https://swent.epfl.ch/">https://swent.epfl.ch/</a></link></li></ul></p>
                                    </div>
                            </div>
                            <div class="course-infos col-xl-4 offset-xl-1 mt-5 mt-lg-0">
                                <div class="study-plans">
                                    <h2 class="h5">In the programs</h2>

                                    <button class="collapse-title collapse-title-desktop " type="button" data-toggle="collapse" data-target="#bama_cyclebachelor-in_1" aria-expanded="true" aria-controls="bama_cyclebachelor-in_1">
                                        Computer Science<br>
                                        <span class="d-block font-weight-normal">2025-2026  Bachelor semester 5</span>
                                    </button>
                                    <div class="collapse-item collapse-item-desktop show" id="bama_cyclebachelor-in_1" style="">
                                        <ul class="list-bullet">
                                                <li><strong>Semester:</strong> Fall</li>
                                                <li><strong>Exam form:</strong> During the semester (winter session)</li>
                                                <li><strong>Subject examined:</strong> The software enterprise - from ideas to products</li>
                                                <li><strong>Courses:</strong> 2 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Exercises:</strong> 1 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Project:</strong> 10 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Type:</strong> optional</li>
                                        </ul>
                                    </div>

                                    <button class="collapse-title collapse-title-desktop  collapsed" type="button" data-toggle="collapse" data-target="#bama_cyclebachelor-sc_2" aria-expanded="false" aria-controls="bama_cyclebachelor-sc_2">
                                        Communication Systems<br>
                                        <span class="d-block font-weight-normal">2025-2026  Bachelor semester 5</span>
                                    </button>
                                    <div class="collapse-item collapse-item-desktop collapse" id="bama_cyclebachelor-sc_2" style="">
                                        <ul class="list-bullet">
                                                <li><strong>Semester:</strong> Fall</li>
                                                <li><strong>Exam form:</strong> During the semester (winter session)</li>
                                                <li><strong>Subject examined:</strong> The software enterprise - from ideas to products</li>
                                                <li><strong>Courses:</strong> 2 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Exercises:</strong> 1 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Project:</strong> 10 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Type:</strong> optional</li>
                                        </ul>
                                    </div>

                                    <button class="collapse-title collapse-title-desktop  collapsed" type="button" data-toggle="collapse" data-target="#min-min_in_3" aria-expanded="false" aria-controls="min-min_in_3">
                                        Computer science minor<br>
                                        <span class="d-block font-weight-normal">2025-2026  Autumn semester</span>
                                    </button>
                                    <div class="collapse-item collapse-item-desktop collapse" id="min-min_in_3" style="">
                                        <ul class="list-bullet">
                                                <li><strong>Semester:</strong> Fall</li>
                                                <li><strong>Exam form:</strong> During the semester (winter session)</li>
                                                <li><strong>Subject examined:</strong> The software enterprise - from ideas to products</li>
                                                <li><strong>Courses:</strong> 2 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Exercises:</strong> 1 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Project:</strong> 10 Hour(s) per week x 14 weeks</li>
                                                <li><strong>Type:</strong> optional</li>
                                        </ul>
                                    </div>
                                </div>

                                <div class="course-schedule mt-5">
                                    <h2 class="h5">Reference week</h2>
                                    <table aria-hidden="true" class="semaineDeRef"><tbody><tr><td class='time'>&nbsp;</td><td>Mo</td><td>Tu</td><td>We</td><td>Th</td><td>Fr</td></tr><tr><td class="time">8-9</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">9-10</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td class="pt-4 pb-3 taken cours" rowspan="2"><a href="https://plan.epfl.ch/?room==CM 1 2">CM2</a></td></tr><tr><td class="time">10-11</td><td class="pt-4 pb-3 taken cours" rowspan="2"><a href="https://plan.epfl.ch/?room==CM 1 1">CM1</a></td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">11-12</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td class="pt-4 pb-3 taken exercice"><a href="https://plan.epfl.ch/?room==CM 1 2">CM2</a></td></tr><tr><td class="time">12-13</td><td class="pt-4 pb-3 taken exercice"><a href="https://plan.epfl.ch/?room==CM 1 1">CM1</a></td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">13-14</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">14-15</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">15-16</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">16-17</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">17-18</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">18-19</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">19-20</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">20-21</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr><tr><td class="time">21-22</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table>
                                    <div class="coursebook-week-caption mt-4" role="presentation" aria-hidden="true">
                                        <p>Légendes:</p>
                                        <p class="caption caption-cours">
                                            <span class="caption-icon"></span>
                                            Lecture
                                        </p>
                                        <p class="caption caption-exercice">
                                            <span class="caption-icon"></span>
                                            Exercise, TP
                                        </p>
                                        <p class="caption caption-projet">
                                            <span class="caption-icon"></span>
                                            Project, Lab, other
                                        </p>
                                    </div>

                                    <!-- accessible content for screen readers -->
                                    <div class="coursebook-week-caption sr-only"><p>Monday, 10h -&nbsp;12h: Lecture <a href="https://plan.epfl.ch/?room==CM 1 1">CM1</a></p><p>Monday, 12h -&nbsp;13h:  Exercise, TP <a href="https://plan.epfl.ch/?room==CM 1 1">CM1</a></p><p>Friday, 9h -&nbsp;11h: Lecture <a href="https://plan.epfl.ch/?room==CM 1 2">CM2</a></p><p>Friday, 11h -&nbsp;12h:  Exercise, TP <a href="https://plan.epfl.ch/?room==CM 1 2">CM2</a></p></div>
                                </div>

                            <div class="my-5" id="relatedcourses">
                                <h2 class="h5 mb-2">Related courses</h2>
                                <p class="text-muted text-sm font-italic mb-4">Results from
                                    <a href="https://graphsearch.epfl.ch">graphsearch.epfl.ch</a>.
                                </p>
                                <graph-widget-course-courses
                                    course-id="CS-311"
                                    locale="en"
                                    limit="5"
                                >
                                </graph-widget-course-courses>
                            </div>

                            </div>
                        </div>

                    </div>
                    <!-- footer -->
                </main>
            </div>
        </div>
        <!-- footer -->
        <div class="bg-gray-100 pt-5">
            <div class="container">
<footer class="footer-light">
    <div class="row">
        <div class="col-6 mx-auto mx-md-0 mb-4 col-md-3 col-lg-2">
            <a href="//www.epfl.ch">
                <img src="https://web2018.epfl.ch/8.1.0/icons/epfl-logo.svg"
                     alt="Logo EPFL, Ecole polytechnique fédérale de Lausanne" class="img-fluid">
            </a>
        </div>
        <div class="col-md-9 col-lg-10 mb-4">
            <div class="ml-md-2 ml-lg-5">
                <ul class="list-inline list-unstyled">
                    <li class="list-inline-item">Contact</li>
                    <li class="list-inline-item text-muted pl-3"><small>EPFL CH-1015 Lausanne</small>
                    </li>
                    <li class="list-inline-item text-muted pl-3"><small>+41 21 693 11 11</small></li>
                </ul>
                <div class="footer-light-socials">
                    <p class="footer-title footer-title-no-underline">Follow the pulses of EPFL on social networks</p>
                    <ul class="social-icon-list list-inline">
                        <li>
                            <a href="https://www.facebook.com/epflcampus"
                            class="social-icon social-icon-facebook social-icon-negative" rel="noopener" target="_blank">
                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-facebook"></use></svg>
                                <span class="sr-only">Follow us on Facebook.</span>
                            </a>
                        </li>
                        <li>
                            <a href="https://instagram.com/epflcampus"
                            class="social-icon social-icon-instagram social-icon-negative" rel="noopener" target="_blank">
                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-instagram"></use></svg>
                                <span class="sr-only">Follow us on Instagram.</span>
                            </a>
                        </li>
                        <li>
                            <a href="https://www.linkedin.com/school/epfl/"
                            class="social-icon social-icon-linkedin social-icon-negative" rel="noopener" target="_blank">
                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-linkedin"></use></svg>
                                <span class="sr-only">Follow us on LinkedIn.</span>
                            </a>
                        </li>
                        <li>
                            <a href="https://social.epfl.ch/@epfl/" 
                            class="social-icon social-icon-mastodon social-icon-negative" rel="noopener" target="_blank">
                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-mastodon"></use></svg>
                                <span class="sr-only">Follow us on Mastodon.</span>
                            </a>
                        </li>
                        <li>
                            <a href="https://x.com/epfl" class="social-icon social-icon-x social-icon-negative"
                            rel="noopener" target="_blank">
                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-x"></use></svg>
                                <span class="sr-only">Follow us on X.</span>
                            </a>
                        </li>
                        <li>
                            <a href="https://www.youtube.com/user/epflnews"
                            class="social-icon social-icon-youtube social-icon-negative" rel="noopener" target="_blank">
                                <svg class="icon" aria-hidden="true"><use xlink:href="#icon-youtube"></use></svg>
                                <span class="sr-only">Follow us on Youtube.</span>
                            </a>
                        </li>
                    </ul>
                </div>
                <div class="footer-legal">
                    <div class="footer-legal-links">
                        <a href="https://www.epfl.ch/about/overview/regulations-and-guidelines/disclaimer/">Accessibility</a>
                        <a href="https://www.epfl.ch/about/overview/regulations-and-guidelines/disclaimer/">Legal notice</a>
                        <a href="https://go.epfl.ch/privacy-policy/">Privacy Policy</a>
                    </div>
                    <div>
                        <p>© 2025 EPFL, all rights reserved</p>
                    </div>
                </div>
            </div>
        </div>
    </div>
</footer>
<button id="back-to-top" class="btn btn-primary btn-back-to-top">
    <span class="sr-only">Back to top</span>
    <svg class="icon" aria-hidden="true">
        <use xlink:href="#icon-chevron-top"></use>
    </svg>
</button>            </div>
        </div>

    </div>
</div>
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-PJGBG5R"
    height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>

<script>
    svgPath = 'https://web2018.epfl.ch/8.1.0/icons/icons.svg';
    featherSvgPath = 'https://web2018.epfl.ch/8.1.0/icons/feather-sprite.svg';
</script>
  
<script src="https://web2018.epfl.ch/8.1.0/js/elements.min.js"></script><script src="https://cdn.jsdelivr.net/gh/epflgraph/graphsearch_widgets@v1.0.2/dist/index.js" type="module"></script>
<script>
  // Hide Related courses if errors.
  jQuery(function(){
    setTimeout(function() {
      const sr = jQuery('graph-widget-course-courses')[0].shadowRoot
      const l = jQuery(sr).find('graph-widget-section').length
      if (l === 0) {
        jQuery('#relatedcourses').hide();
      }
    }, 1000)
  });
</script>
</body>
</html>