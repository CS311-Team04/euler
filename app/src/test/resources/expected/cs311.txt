CS-311 – The Software Enterprise
Course information
Semester: Autumn
Credits: 4
Lecturer: Professor Bertrand Meyer
Language: English
Category: Software engineering

Course description
This course introduces students to software project management, agile development, and teamwork in a simulated enterprise environment. Students collaborate in teams to design, implement, and deliver a software product.

Objectives
– Understand software project planning and organization
– Apply agile and iterative development practices
– Manage team collaboration using version control
– Learn communication and documentation practices in an enterprise context

Prerequisites
Basic knowledge of programming and software engineering principles.

Evaluation
Assessment is based on project work, team reports, and presentations. Attendance and participation are required.

Learning outcomes
Students gain practical experience in team-based software development, project delivery, and communication skills relevant to professional environments.
